### PR TITLE
Use coroutines instead of asynctask in DownloadFormListTask

### LINKED ISSUE #5001: Use coroutines instead of asynctask in ResetDialogPreferenceFragmentCompat
#### Problem description
We have multiple old asynctasks across the project. As you know it's better to use coroutines and asynctask is now even deprecated.
`ResetDialogPreferenceFragmentCompat` contains a small asynctask and replacing it should be a good first step.

### PR DESCRIPTION
**Pull request by dczerwinski**
Closes #5001 

#### What has been done to verify that this works as intended?
Tested manually.

#### Why is this the best possible solution? Were any other approaches considered?
Async tasks are deprected. I replaced them with Corutines.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Should not affect users.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [X] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [X] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [X] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.kt [hunk 1]
```diff
@@ -0,0 +1,708 @@
...
+ * and you'll have to hit 'refresh' where it will ask for credentials again. Technically a server
+ * could point at other servers requiring authentication to download the forms, but the current
+ * implementation in Collect doesn't allow for that. Mostly this is just because it's a pain in the
+ * butt to keep track of which forms we've downloaded and where we're needing to authenticate. I
+ * think we do something similar in the instanceuploader task/activity, so should change the
+ * implementation eventually.
+ *
+ * @author Carl Hartung (carlhartung@gmail.com)
+ */
+class FormDownloadListActivity :
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.kt]: Usually I would encourage you to convert classes that you touch to kotlin but in this case I feel uneasy. The class is big and I think it makes it risky. Please just focus on replacing asynctask with coroutines and revert this change.

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]

File: collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.kt (hunk 1 above)
Comment (dczerwinski) [reply collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.kt]: Done. I moved logic of task to viewModel and operate there with coroutines and livedata.

**Comment by seadowg** [comment]
Closing this for the moment but we should come back to it as a reference when we're looking at the relevant code again.
